src/components/organisms/History.tsx

```diff
@@ -71,39 +71,55 @@ export default function History() {
       },
     };
 
+    // TODO
     const res = await fetch(
       `${url}/device-data/${device_id}/historical/week`,
       options
     );
     const data = await res.json();
-
     console.log("data: ", data);
+
+    const tempMax: number[] = [];
+    const tempMin: number[] = [];
+    const humidMax: number[] = [];
+    const humidMin: number[] = [];
+    const label: string[] = [];
+
+    data.forEach((day: any) => {
+      tempMax.push(day.max_temp);
+      tempMin.push(day.min_temp);
+      humidMax.push(day.max_humid);
+      humidMin.push(day.min_humid);
+      label.push(day.date);
+    });
+
+    setDataTempMax(tempMax);
+    setDataTempMin(tempMin);
+    setDataHumidMax(humidMax);
+    setDataHumidMin(humidMin);
+    setDataLabels(label);
   };
 
+  useEffect(() => {
+    weekButtonHandler();
+  }, []);
+
   // TODO change to actual dates from API - maybe convert dates to weekday
-  const labels = [
-    "Monday",
-    "Tuesday",
-    "Wednesday",
-    "Thursday",
-    "Friday",
-    "Saturday",
-    "Sunday",
-  ];
+  const labels = dataLabels;
```

Comment:
Do you have to copy this variable?

Reply:
maybe can use state directly but it needs to be named 'labels' to work in the chart and have kept state variable naming consistent